pkg/provider/azure.go

```diff
@@ -31,6 +31,86 @@ type AzureProvider struct {
 	ctx           map[string]interface{}
 }
 
+var (
+	azureRegions = []string{
+		"East US",
```

Comment:
These are not the correct region codes for Azure.

Reply:
@michaeljguarino You can find the correct ones here. https://azuretracks.com/2021/04/current-azure-region-names-reference/

Reply:
Some of these option are most likely not valid for AKS. We should verify that this list (and maybe the other cloud providers as well) all support the managed Kubernetes services we are using.

Reply:
I think they actually do still work when configured, I got them via: `az account list-locations | jq '.[].displayName'`

Reply:
Azure seems to have some code to coerce them to the downcased/non-whitespaced version of the names